test/unit/models/discordactions.test.js

```diff
@@ -578,6 +591,155 @@ describe("discordactions", function () {
+        userStatusDataForOooState: oooUserStatus,
+      } = userStatusData;
+      const userIdList = await Promise.all([
+        await addUser(idleUser), // idle user with no task progress updates
+        await addUser(activeUserWithProgressUpdates), // active user with task progress updates
+        await addUser(activeUserWithNoUpdates), // active user with no task progress updates
+        await addUser(userNotInDiscord), // OOO user with
+      ]);
+      activeUserId = userIdList[2];
+      await Promise.all([
+        await userStatusModel.updateUserStatus(userIdList[0], idleUserStatus),
+        await userStatusModel.updateUserStatus(userIdList[1], activeUserStatus),
+        await userStatusModel.updateUserStatus(userIdList[2], activeUserStatus),
+        await userStatusModel.updateUserStatus(userIdList[3], oooUserStatus),
+      ]);
+
+      const tasksPromise = [];
+
+      for (let index = 0; index < 4; index++) {
+        const task = tasksData[index];
+        const validTask = {
+          ...task,
+          assignee: userIdList[index],
+          startedOn: (new Date().getTime() - convertDaysToMilliseconds(7)) / 1000,
+          endsOn: (new Date().getTime() + convertDaysToMilliseconds(4)) / 1000,
+          status: TASK_STATUS.IN_PROGRESS,
+        };
+
+        tasksPromise.push(tasksModel.add(validTask));
+      }
+      const taskIdList = (await Promise.all(tasksPromise)).map((tasksDoc) => tasksDoc.id);
+      const progressDataList = [];
+
+      const date = new Date();
+      date.setDate(date.getDate() - 1);
+      const progressData = stubbedModelTaskProgressData(null, taskIdList[2], date.getTime(), date.valueOf());
+      progressDataList.push(progressData);
+
+      await Promise.all(progressDataList.map(async (progress) => await createProgressDocument(progress)));
+      const discordMembers = [...getDiscordMembers];
+      discordMembers[0].roles.push("9876543210");
+      discordMembers[1].roles.push("9876543210");
+      sinon.stub(discordService, "getDiscordMembers").returns(discordMembers);
+    });
+    afterEach(async function () {
+      sinon.restore();
+      await cleanDb();
+    });
+    it("should list of users who missed updating progress", async function () {
+      const result = await getMissedProgressUpdatesUsers();
+      expect(result).to.be.an("object");
+      expect(result).to.be.deep.equal({
+        tasks: 4,
+        missedUpdatesTasks: 3,
+        usersToAddRole: [activeUserWithProgressUpdates.discordId],
+      });
+    });
+
+    it("should not list of users who are not active and who missed updating progress", async function () {
+      const result = await getMissedProgressUpdatesUsers();
+      expect(result).to.be.an("object");
+      expect(result.usersToAddRole).to.not.contain(idleUser.discordId);
+      expect(result.usersToAddRole).to.not.contain(userNotInDiscord.discordId);
+    });
+
+    it("should not list of users when exception days are added", async function () {
```

Comment:
In actual is there a way how super user will specify `exception days` while running the script

Reply:
Yes by passing inputs as query params in api.